crates/router/src/db/locker_mock_up.rs

```diff
@@ -62,25 +62,180 @@ impl LockerMockUpInterface for Store {
     }
 
     async fn insert_locker_mock_up(
         &self,
-        _new: storage::LockerMockUpNew,
+        new: storage::LockerMockUpNew,
     ) -> CustomResult<storage::LockerMockUp, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut locked_lockers = self.lockers.lock().await;
+
+        if locked_lockers.iter().any(|l| l.card_id == new.card_id) {
+            Err(errors::StorageError::MockDbError)?;
+        }
+
+        let created_locker = storage::LockerMockUp {
+            #[allow(clippy::as_conversions)]
+            id: locked_lockers.len() as i32,
+            card_id: new.card_id,
+            external_id: new.external_id,
+            card_fingerprint: new.card_fingerprint,
+            card_global_fingerprint: new.card_global_fingerprint,
+            merchant_id: new.merchant_id,
+            card_number: new.card_number,
+            card_exp_year: new.card_exp_year,
+            card_exp_month: new.card_exp_month,
+            name_on_card: new.name_on_card,
+            nickname: None,
+            customer_id: new.customer_id,
+            duplicate: None,
+            card_cvc: new.card_cvc,
+            payment_method_id: new.payment_method_id,
+        };
+
+        locked_lockers.push(created_locker.clone());
+
+        Ok(created_locker)
     }
 
     async fn delete_locker_mock_up(
         &self,
-        _card_id: &str,
+        card_id: &str,
     ) -> CustomResult<storage::LockerMockUp, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut locked_lockers = self.lockers.lock().await;
+
+        let position = locked_lockers
+            .iter()
+            .position(|l| l.card_id == card_id)
+            .ok_or(errors::StorageError::MockDbError)?;
+
+        Ok(locked_lockers.remove(position))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[allow(clippy::unwrap_used)]
+    mod mockdb_locker_mock_up_interface {
+        use crate::{
+            db::{locker_mock_up::LockerMockUpInterface, MockDb},
+            types::storage,
+        };
+
+        pub struct LockerMockUpIds {
+            card_id: String,
+            external_id: String,
+            merchant_id: String,
+            customer_id: String,
+        }
+
+        fn create_locker_mock_up_new(locker_ids: LockerMockUpIds) -> storage::LockerMockUpNew {
+            storage::LockerMockUpNew {
+                card_id: locker_ids.card_id,
+                external_id: locker_ids.external_id,
+                card_fingerprint: "card_fingerprint".into(),
+                card_global_fingerprint: "card_global_fingerprint".into(),
+                merchant_id: locker_ids.merchant_id,
+                card_number: "card_number".into(),
+                card_exp_year: "card_number".into(),
+                card_exp_month: "card_number".into(),
```

Comment:
```suggestion
                card_number: "card_number".into(),
                card_exp_year: "card_exp_year".into(),
                card_exp_month: "card_exp_month".into(),
```

Personally, I'd prefer having a 16 digit card number, 4 digit expiry year, 2 digit expiry month and a 3 digit CVC passed instead of these placeholders.